### DISCUSSION TITLE
Add Godot resource and/or scene file export

### DISCUSSION THREAD

**Discussion by henlo-birb** [Orama-Interactive/Pixelorama#566] [Ideas]
Since Pixelorama is built using Godot, it would be relatively (compared to being built in some other framework) easy to add options to export Godot resource or scene files, which would greatly streamline the process of adding sprites to a project.

**Comment by OverloadedOrama** [comment]
This is a nice idea. I'm not exactly sure how this can be implemented though. The idea is, we export a scene file and its children will be **Sprite** nodes for every frame?

Feel free to share more ideas and more details about this.

**Comment by henlo-birb** [comment]
We could have options to export to a number of different godot formats. All of the kinds of nodes that would take the animations take them as _resources_. I think the resources that would make sense to support here would be ImageTexture, AnimatedTexture, SpriteFrames, and TileSet. Maybe a far off  version with tools for theme creating? Png files work as ImageTexture, so that's already done. AnimatedTexture is the easiest of the rest to implement, as it takes a bunch of frames as Textures, an fps, and optionally, extra delay times for individual frames, which is probably ok to ignore for the time being. SpriteFrames and TileSet are doable, but would involve a lot of work to implement, so those would be a long-term goal.

Adding Sprite and AnimatedSprite node exports would probably mean adding a stripped-down version of the inspector for those nodes, along with whatever settings are involved in generating the resources you want to use with them. 

Both the resources and the nodes can be saved using ResourceSaver, although the nodes must be first converted to PackedScene.

**Comment by lmerriam** [comment]
Would absolutely love to see this, heavy integration with Godot (perhaps via a Godot addon) would make Pixelorama a must-have for a dev's workflow.

Would be great to be able to edit and sync sprite resources and animation properties between the programs, similar to how Adobe programs all interface with each other and sync changes automatically without requiring import/exports.

**Comment by silentorb** [comment]
There's now a plugin to import Pixelorama projects into Godot: https://github.com/Technohacker/godot_pixelorama_importer.
